iwyu_include_picker.cc

```diff
@@ -1247,16 +1248,42 @@ IncludePicker::IncludePicker(RegexDialect regex_dialect,
 
 void IncludePicker::AddDefaultMappings(CStdLib cstdlib,
                                        CXXStdLib cxxstdlib) {
+  using clang::tooling::stdlib::Header;
+  using clang::tooling::stdlib::Lang;
+  using clang::tooling::stdlib::Symbol;
+
   if (cstdlib == CStdLib::Glibc) {
     AddSymbolMappings(libc_symbol_map, IWYU_ARRAYSIZE(libc_symbol_map));
     AddIncludeMappings(libc_include_map, IWYU_ARRAYSIZE(libc_include_map));
+  } else if (cstdlib == CStdLib::Canonical) {
+    // Get canonical C standard library mappings from clang tooling
+    for (auto& sym : Symbol::all(Lang::C)) {
+      string name = sym.name().str();
+      auto headers = sym.headers();
+
+      // the canonical header is returned first
+      for (auto& header : headers) {
+        AddSymbolMapping(name, MappedInclude(header.name().str()), kPublic);
+      }
+    }
   }
 
   if (cxxstdlib == CXXStdLib::Libstdcxx) {
     AddSymbolMappings(libstdcpp_symbol_map,
                       IWYU_ARRAYSIZE(libstdcpp_symbol_map));
     AddIncludeMappings(libstdcpp_include_map,
                        IWYU_ARRAYSIZE(libstdcpp_include_map));
+  } else if (cxxstdlib == CXXStdLib::Canonical) {
+    // Get canonical C++ standard library mappings from clang tooling
+    for (auto& sym : Symbol::all(Lang::CXX)) {
```

Comment:
For all three `auto`-s here -- we only use `auto` when the type is clearly visible on the same line, so let's spell out the type.